README.md

```diff
@@ -48,69 +98,375 @@ These bindings allow Azure Functions to integrate with [Azure SignalR Service](h
+
+```cs
+[FunctionName("negotiate")]
+public static SignalRConnectionInfo Negotiate(
+    [HttpTrigger(AuthorizationLevel.Anonymous)]HttpRequest req, 
+    [SignalRConnectionInfo
+        (HubName = "chat", UserId = "{headers.x-ms-client-principal-id}")]
+        SignalRConnectionInfo connectionInfo)
 {
-  "type": "signalRConnectionInfo",
-  "name": "connectionInfo",
-  "hubName": "<hub_name>",
-  "connectionStringSetting": "<setting_name>", // Defaults to AzureSignalRConnectionString
-  "direction": "in"
+    // connectionInfo contains an access key token with a name identifier claim set to the authenticated user
+    return connectionInfo;
 }
 ```
 
-### Using the SignalR output binding
+### 2.x JavaScript input examples
 
-The `SignalR` output binding can be used to broadcast messages to all clients connected a hub. Take a look at this sample:
+The following example shows a SignalR connection info input binding in a *function.json* file and a [JavaScript function](https://docs.microsoft.com/en-us/azure/azure-functions/functions-reference-node) that uses the binding to return the connection information.
 
-- [HttpTrigger function to send messages](samples/simple-chat/js/functionapp/messages/)
-- [Simple chat app](samples/simple-chat/content/index.html)
-    - Calls negotiate endpoint to fetch connection information
-    - Connects to SignalR Service
-    - Sends messages to HttpTrigger function, which then broadcasts the messages to all clients
+Here's binding data in the *function.json* file:
 
-Binding schema:
+Example function.json:
+
+```json
+{
+    "type": "signalRConnectionInfo",
+    "name": "connectionInfo",
+    "hubName": "chat",
+    "connectionStringSetting": "<name of setting containing SignalR Service connection string>",
+    "direction": "in"
+}
+```
+
+Here's the JavaScript code:
 
 ```javascript
+module.exports = async function (context, req, connectionInfo) {
+    context.res.body = connectionInfo;
+};
+```
+
+#### Authenticated tokens
+
+If the function is triggered by an authenticated client, you can add a user ID claim to the generated token. You can easily add authentication to a function app using [App Service Authentication](https://docs.microsoft.com/en-us/azure/app-service/overview-authentication-authorization).
+
+App Service Authentication sets HTTP headers named `x-ms-client-principal-id` and `x-ms-client-principal-name` that contain the authenticated user's client principal ID and name, respectively. You can set the `userId` property of the binding to the value from either header using a [binding expression](https://docs.microsoft.com/en-us/azure/azure-functions/functions-bindings-expressions-patterns): `{headers.x-ms-client-principal-id}` or `{headers.x-ms-client-principal-name}`. 
```

Comment:
duplicated content with before, move to parent title descriptions.

Reply:
level issue, it is in each 2.x [language] input examples

Reply:
Yes, understand. But it's common for different languages, so how about move to upper level for a general explanation.

Reply:
but the way to handle the token is different accross languages